pkg/pkg.bzl

```diff
@@ -142,7 +162,16 @@ def _pkg_tar_impl(ctx):
         },
         use_default_shell_env = True,
     )
-    return OutputGroupInfo(out = [ctx.outputs.out])
+    return [
+        DefaultInfo(
+            files = depset([output_file]),
+            runfiles = ctx.runfiles(files = [output_file]),
+        ),
+        PackageArtifactInfo(
+            label = ctx.label.name,
+            file_name = output_name,
+        )
```

Comment:
I find this to be somewhat clumsy, but I simultaneously can't think of anything better.  So, to actually get the proper name of the package to include in some output, I now need to say something like this:

```python
files = []
for f in ctx.files.srcs:
    if f.owner != None and PackageArtifactInfo in f.owner:
        files.append(f[PackageArtifactInfo].file_name)
    else:
        files.append(f)
```

Perhaps something like this should be in a helper function?

Reply:
I could create a helper when I expand this to the other packages.

To be fair. I am hard pressed to find cases where I need to know the full path.
Do you have a specific use case in mind?
Could that case also be solved when we allow package_variables expansion into .spec and manifest files?

Reply:
(I'll be working on the other comments tomorrow, this comes to mind now)

> To be fair. I am hard pressed to find cases where I need to know the full path.
> Do you have a specific use case in mind

One use case is when you have to include package outputs in some other aggregate (a package, or `filegroup`).  Say you have package $A, which has a templated name ($A_t), and some aggregate $B, which needs to contain one of $A or $A_t.  How does one ensure that this happens?

It's clear what to do when you have access to the rule sources, but what if you can't modify them?

>  Could that case also be solved when we allow package_variables expansion into .spec and manifest files?

Perhaps, but it's inconvenient to have to respecify the precise template to get what the named artifact.  It's better to have the desired artifact identifiable by an alias somehow.  Perhaps there could be a conversion of sorts between `PackageArtifactInfo` and `PackageVariableInfo`?

This doesn't help with things like `filegroup`s or native rules, though.

Reply:
The BUILD file is going to have to look something like 

```
pkg_tar(name="foo", package_file_name="custom_name.tar", ...)

pkg_tar(name="outer", srcs=[:foo, README.md])
```

outer must reference "foo" because that is the name of the target, but want's  "custom_name.tar" to be in the included file.  Right now I have the PackageArtifactInfo.  That would work but you have to look for it specially.

Alternately, we could emit a PackageFilegroupInfo provider, where the src was the target ":foo" and the path contained the name. That probably composes with the other rules better. But we would need the pkg_filegroup provider first. I'll happily switch to that once we have it in place.  Like I said in a PR a few days ago, I'ld like to see us get the provider defined, even if we don't have a lot of rules that produce or consume it.

Reply:
This sounds good enough for now, but I think this could be worth iterating upon some more.